src/lib.rs

```diff
@@ -99,24 +99,90 @@ impl PDBCache {
 /// # Safety
 /// This function is unsafe because it uses the Windows API.
 /// Do not call this function in a threaded context.
-unsafe fn inject(
-    function_name: &str,
-    hook: unsafe fn(u64) -> Result<(), rivets::retour::Error>,
-) -> Result<()> {
+unsafe fn inject(hook: &RivetsHook) -> Result<()> {
     let factorio_path = std::path::Path::new("C:/Users/zacha/Documents/factorio/bin");
 
-    let Some(address) = PDBCache::get(factorio_path)?.get_function_address(function_name) else {
-        bail!("Failed to find {function_name} address");
+    let Some(address) =
+        PDBCache::get(factorio_path)?.get_function_address(hook.mangled_name.as_str())
+    else {
+        bail!("Failed to find {} address", hook.mangled_name);
     };
-    println!("{function_name} address: {address:#x}");
 
-    unsafe { Ok(hook(address)?) }
+    (hook.hook)(address)
+        .into_result()
+        .map_err(std::convert::Into::into)
+}
+
+fn extract_all_mods_libs(
+    read_data: impl AsRef<Path>,
+    write_data: impl AsRef<Path>,
+) -> Result<Vec<PathBuf>> {
+    #[cfg(target_os = "linux")]
+    static DYNAMIC_LIBRARY_SUFFIX: &str = ".so";
+    #[cfg(target_os = "linux")]
+    static RIVETS_LIB: &str = "rivets.so";
+    #[cfg(target_os = "windows")]
+    static DYNAMIC_LIBRARY_SUFFIX: &str = ".dll";
+    #[cfg(target_os = "windows")]
+    static RIVETS_LIB: &str = "rivets.dll";
+
+    let mut result = vec![];
+    let mod_list = ModList::generate_custom(read_data, &write_data)?;
```

Comment:
Fixing the issue that caused injector to not find mods. Also run `cargo update` to get an updated version of `mod_util`
```suggestion
    let mut mod_list = ModList::generate_custom(read_data, &write_data)?;
    mod_list.load();
```